src/error.rs

```diff
@@ -96,6 +116,51 @@ pub struct Transport {
     source: Option<Box<dyn error::Error + Send + Sync + 'static>>,
 }
 
+impl Transport {
+    /// The type of error that happened while processing the request.
+    pub fn kind(&self) -> ErrorKind {
+        self.kind
+    }
+
+    /// Higher level error details, if there are any.
+    ///
+    /// ```
+    /// use ureq::ErrorKind;
+    /// use std::error::Error;
+    /// use url::ParseError;
+    ///
+    /// let result = ureq::get("broken/url").call();
+    /// let error = result.unwrap_err().into_transport().unwrap();
+    ///
+    /// // the display trait is a combo of the underlying classifications
+    /// assert_eq!(error.to_string(),
+    ///     "Bad URL: failed to parse URL: RelativeUrlWithoutBase: relative URL without a base");
+    ///
+    /// // classification
+    /// assert_eq!(error.kind(), ErrorKind::InvalidUrl);
+    /// assert_eq!(error.kind().to_string(), "Bad URL");
+
+    /// // higher level message
+    /// assert_eq!(error.message(), Some("failed to parse URL: RelativeUrlWithoutBase"));
+    ///
+    /// // boxed underlying error
+    /// let source = error.source().unwrap();
+    /// // downcast to original error
+    /// let downcast = source.downcast_ref::<ParseError>().unwrap();
+    ///
+    /// assert_eq!(downcast.to_string(), "relative URL without a base");
+    /// ```
```

Comment:
This documentation is nice to have on the Transport type, but it's confusing to have it on the `message` method since it's about `.source()` (and downcast_ref). Maybe move it up to the `Transport` struct itself?

Reply:
Will do!